### PR TITLE
Added unit test for calculate fit size

### LINKED ISSUE #598: Write unit test for calculateFitSize method in ImageUtils
The `calculateFitSize` method in ImageUtils is untested. It would be great if unit tests could be added with various test cases in order to test if the method works as expected.

Please write tests for the method with various input arguments.

### PR DESCRIPTION
**Pull request by jddeep**
# Description
Added unit test for calculate fit size.
Fixes #598 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by jddeep** [comment]
@srv-twry @Swati4star  Please review.

File: app/src/test/java/swati4star/createpdf/util/ImageUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.junit.Test;
+
+public class ImageUtilsTest {
+
+    @Test
+    public void testcalculateFitSize() {
+        float testWidth = 8.0f;
+        float testHeight = 12.0f;
+        Rectangle testDocumentSize = new Rectangle(5.0f, 5.0f, 5.0f, 5.0f);
```
Comment (Swati4star) [inline app/src/test/java/swati4star/createpdf/util/ImageUtilsTest.java]: Add test case for 
- negative points
- invalid length (length < 0)
- rectangle is a point
- rectangle is a line

**Comment by jddeep** [comment]
@Swati4star Changes have been seen into. Please review

File: app/src/test/java/swati4star/createpdf/util/ImageUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        float testWidthThree = 8.0f;
+        float testHeightThree = 12.0f;
+        Rectangle testDocumentSizePoint = new Rectangle(0.0f, 0.0f, 0.0f, 0.0f);
+        new ImageUtils().calculateFitSize(testWidthThree, testHeightThree, testDocumentSizePoint);
+
+        float testWidthFour = 8.0f;
+        float testHeightFour = 12.0f;
+        Rectangle testDocumentSizeLine = new Rectangle(5.0f, 5.0f, 0.0f, 0.0f);
+        new ImageUtils().calculateFitSize(testWidthFour, testHeightFour, testDocumentSizeLine);
```
Comment (Swati4star) [inline app/src/test/java/swati4star/createpdf/util/ImageUtilsTest.java]: You are not supposed to just call up the function. You need to compare it with actual value also 

For reference, check this commit https://github.com/Swati4star/Images-to-PDF/commit/f14f2f07872c4d0ae2cd3ae16cf8d954160e0365

File: app/src/main/java/swati4star/createpdf/util/ImageUtils.java [hunk 3]
```diff
@@ -34,12 +34,12 @@
     /**
      * Calculates the optimum size for an image, such that it scales to fit whilst retaining its aspect ratio
      *
-     * @param originalWidth the original width of the image
+     * @param originalWidth  the original width of the image
      * @param originalHeight the original height of the image
-     * @param documentSize a rectangle specifying the width and height that the image must fit within
+     * @param documentSize   a rectangle specifying the width and height that the image must fit within
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/util/ImageUtils.java]: what is the point of these unnecessary changes? Revert them.

File: app/src/main/java/swati4star/createpdf/util/ImageUtils.java [hunk 4]
```diff
@@ -34,12 +34,12 @@
...
      * @return a rectangle that provides the scaled width and height of the image
      */
-    public static Rectangle calculateFitSize(float originalWidth, float originalHeight, Rectangle documentSize) {
+    public Rectangle calculateFitSize(float originalWidth, float originalHeight, Rectangle documentSize) {
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/util/ImageUtils.java]: why did you make it non-static?

**Comment by Swati4star** [comment]
@jddeep The tests you have added are failing. check the travis link

**Comment by jddeep** [comment]
Screenshot (145)
@Swati4star I am having this kind of error. Can you help me out here? Actually, I am new to adding tests. Your help here would be much appreciated. Thanks.

**Comment by Swati4star** [comment]
calculateFitSize is returning Rectangle, but you are comparing it with integer.

**Comment by jddeep** [comment]
@Swati4star I have made some changes, please take a look.

**Comment by jddeep** [comment]
Screenshot (146)

@Swati4star Check this out!

**Comment by srv-twry** [comment]
@jddeep You're comparing floating point values that's why you're getting this issue. It may be possible that exact representation of that number doesn't exist using the number of bits used in the number. In short you can't really compare floats using `==` operator.

The usual way to do it to define a very small value named `eps` and set it to `1e-9`. Then check if `abs(number - 0.5f) < eps`. If yes, then they are equal.

**Comment by jddeep** [comment]
@Swati4star Changes have been updated. Please review. Thanks.

File: app/src/test/java/swati4star/createpdf/util/ImageUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class ImageUtilsTest {
+    private static final Rectangle NO_NORMAL_ERROR =
+            new Rectangle(0.0f, 0.0f, 0.0f, 0.0f);
+    private static final Rectangle NO_POINT_ERROR =
+            new Rectangle(0.0f, 0.0f, 0.0f, 0.0f);
+    private static final Rectangle NO_LINE_ERROR =
+            new Rectangle(0.0f, 0.0f, 0.0f, 0.0f);
```
Comment (srv-twry) [inline app/src/test/java/swati4star/createpdf/util/ImageUtilsTest.java]: I am not able to understand what these three rectangle's mean especially since they have everything equal to 0.0f. Can you explain please?

Comment (jddeep) [reply app/src/test/java/swati4star/createpdf/util/ImageUtilsTest.java]: They are just to signify different errors to different comparisions. It can be done with a single `NO ERROR` variable too.
